### ISSUE TITLE
Failed to AOT compile Microsoft.Graph.dll, the AOT compiler exited with code 134

### ISSUE THREAD

**Issue by NickDarvey** [xamarin/xamarin-macios#18388]
### Steps to Reproduce

1. Clone https://github.com/microsoftgraph/msgraph-sample-maui
2. Remove [bundle resource 'appSettings.Development.json'](https://github.com/microsoftgraph/msgraph-sample-maui/blob/cf05d8540e82e9e9c1b346a3ba8332a6b68c2a69/GraphMAUI/GraphMAUI.csproj#L87) from project
3. Run `dotnet publish -c Release -f net7.0-ios16.1 -r ios-arm64`
4. Wait ~1 hour because of https://github.com/microsoftgraph/msgraph-sample-maui/issues/68 

### Expected Behavior

App is published.

### Actual Behavior

```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.2.1024/targets/Xamarin.Shared.Sdk.targets(1011,3): Failed to AOT compile Microsoft.Graph.dll, the AOT compiler exited with code 134
```

If I use `<UseInterpreter>true</UseInterpreter>`, the build succeeds.

### Environment

- dotnet 7.0.100
- Microsoft.iOS.Sdk 16.2.1024

### Build Logs

[msbuild_(6).zip](https://github.com/xamarin/xamarin-macios/files/11619728/msbuild_.6.zip)

### Example Project

https://github.com/microsoftgraph/msgraph-sample-maui

**Comment by NickDarvey** [comment]
> Wait ~1 hour because of https://github.com/microsoftgraph/msgraph-sample-maui/issues/68

I'm trying to work around this issue so that we can at least reach the AOT compilation error faster.

Almost all of this 1 hour is spent in the `Microsoft.NET.ILLink.targets/ILLink` task.

The Microsoft.Graph.dll is only copied/skipped by the linker, but if I remove it from `ManagedAssemblyToLink` (the argument to ILLink's `AssemblyPaths` parameter) the time drops to ~10 minutes.

```xml
<ItemGroup>
    <UnanalyzableAssembly Include="Microsoft.Graph" />
</ItemGroup>

<Target Name="ConfigureTrimming"
    BeforeTargets="PrepareForILLink">
    <ItemGroup>
        <_UnanalyzableManagedAssemblyToLink Include="@(UnanalyzableAssembly)" NuGetPackageId="%(Identity)" />
        <ManagedAssemblyToLink Remove="@(_UnanalyzableManagedAssemblyToLink)" MatchOnMetadata="NuGetPackageId" />
    </ItemGroup>
</Target>
```

However doing so then causes the AOT compilation to fail on the app instead.

```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.2.1024/targets/Xamarin.Shared.Sdk.targets(1011,3): Failed to AOT compile GraphMAUI.dll, the AOT compiler exited with code 139
```

It looks like excluding an assembly from `ManagedAssemblyToLink` causes it to be omitted from [`_AssembliesToAOT`](https://github.com/xamarin/xamarin-macios/blob/2606b2341e1014bdf0549b264fe07d9b572fb6f7/tools/dotnet-linker/Steps/ComputeAOTArguments.cs#L54) and my assumptions is that is what is causing the compiler to fail on GraphMAUI.dll. So, my next step is to see if I can reinstate it after the linker runs but before the AOT compiler runs.

**Comment by Soap-141** [comment]
I'm having the same issue and mtouchInterpreter does nothing

**Comment by rolfbjarne** [comment]
I'm not sure if the resulting app works as expected, but asking the linker to trim Microsoft.Graph.dll makes the build succeed:

```xml
<ItemGroup>
    <TrimmableAssembly Include="Microsoft.Graph" />
</ItemGroup>
```

**Comment by Soap-141** [comment]
For me
* Assertion at /Users/runner/work/1/s/src/mono/mono/mini/mini-llvm.c:6004, condition `lhs' not met

https://github.com/dotnet/runtime/blob/417a01523fb75fa2688b6407db2e71ded7063f72/src/mono/mono/mini/mini-llvm.c#L6004

**Comment by Soap-141** [comment]
@rolfbjarne You think it's a good idea to disable llvm?

**Comment by rolfbjarne** [comment]
@Soap-141 there _might_ be perf hit, but if you don't notice / don't care, then it won't be a problem

**Comment by Soap-141** [comment]
@rolfbjarne I'll test it out to and compare it to UseInterpreter=true

**Comment by rolfbjarne** [comment]
I've filed:

* https://github.com/dotnet/runtime/issues/87061 for the AOT compiler crash
* https://github.com/dotnet/runtime/issues/87062 for the slow linker.

Feel free to CC yourself on these issues to be updated.

I'll also close this issue, since I don't think there's any other issue here relative to this repo.

**Comment by Soap-141** [comment]
> I'm not sure if the resulting app works as expected, but asking the linker to trim Microsoft.Graph.dll makes the build succeed:
> 
> ```
> <ItemGroup>
>     <TrimmableAssembly Include="Microsoft.Graph" />
> </ItemGroup>
> ```

This worked thanks!

**Comment by NickDarvey** [comment]
@rolfbjarne, thanks for investigating.

To summarize:
1. The AOT compiler crashing.
   Fixed in main, so eventually part of dotnet 8.
   https://github.com/dotnet/runtime/pull/85952
   Workaround with `<TrimmableAssembly Include="Microsoft.Graph" />`
2. ILLink being quite slow.
   Already fixed, update to dotnet >7.0.302
   https://github.com/dotnet/runtime/issues/87050#issuecomment-1574346833
